building/qubes-r3-building.md

```diff
@@ -24,36 +38,80 @@ $ wget https://keys.qubes-os.org/keys/qubes-master-signing-key.asc
+
+~~~
+$ wget https://github.com/QubesOS/qubes-builder-debian/blob/master/keys/whonix-developer-patrick.asc
+$ gpg --import whonix-developer-patrick.asc
+~~~
+
+**Note** In the above process, we do *not* rely on the security of our server (keys.qubes-os.org) nor the connection (ssl, cert) -- we only rely on you getting the Qubes Master Signing Key fingerprint *somehow* and ensuring they match!
+Likewise, the Whonix signing key is available from multiple sources.
 
-Now lets bootstrap the builder. Unfortunately the builder cannot verify itself (the classic Chicken and Egg problem), so we need to verify the signature manually:
+Now let's bootstrap the builder. Unfortunately, the builder cannot verify itself (the classic Chicken and Egg problem), so we need to verify the signature manually:
 
 ~~~
 $ git clone git://github.com/QubesOS/qubes-builder.git
+$ mkdir qubes-builder/keyrings
+$ mkdir qubes-builder/keyrings/git
+$ cp .gnupg/pubring.gpg qubes-builder/keyrings/git/
+$ cp .gnupg/trustdb.gpg qubes-builder/keyrings/git/
 $ cd qubes-builder
-$ git describe --exact-match HEAD
-<some tag>
-$ git tag -v <some tag>
+$ git tag -v `git describe`
+~~~
+
+Assuming the verification went fine, we're good to go with all the rest without ever thinking more about verifying digital signatures on all the rest of the components.
+The builder will do that for us for each component, every time we build, even for all auxiliary files (e.g. Xen or Linux kernel sources).
+
+Let's configure the builder first (see Note at bottom if you would prefer to manually configure):
+
+~~~
+$ ./setup
+# Select 3.2 or 4.0 for version
+# Stable
+# No (we want a full build)
+# Select builder-fedora, builder-debian, template-whonix, mgmt-salt
+# Yes (to download)
+# Select fc26, stretch, whonix-gateway, whonix-workstation (for the currently shipping templates)
 ~~~
 
-Assuming the verification went fine, we're good to go with all the rest without ever thinking more about verifying digital signatures on all the rest of the components, as the builder will do that for us, for each component, every time we, even for all aux files (e.g. Xen or Linux kernel sources).
+Once the download is complete, continue the build process with:
+
+~~~
+$ make install-deps
+$ make get-sources
+~~~
 
-Let's configure the builder first (we can use one of the example configs, either for R2 or "master", which currently means pre-released R3):
+Finally, if you are making a test build, use:
 
 ~~~
-cp example-configs/qubes-os-master.conf builder.conf
+$ make qubes
+$ make iso
 ~~~
 
-You can take a loot at the `builder.conf.default` for a description of all available options. Nevertheless, the default config should be enough for start:
+Or for a fully signed build (this requires setting SIGN_KEY in the builder.conf):
 
 ~~~
-$ make get-sources qubes
-$ make sign-all # this requires setting SIGN_KEY in the builder.conf, can be skipped for test builds.
+$ make sign-all
```

Comment:
This suggests that `make qubes` isn't needed for signed build, which isn't true.

Reply:
Original doc omitted that step then.

Reply:
In reverse order, should be:
```
make qubes
make sign-all
make iso
```